nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/IApiHandler.cs

```diff
@@ -35,6 +35,28 @@ public static async Task RecordUpdateJobError(this IApiHandler handler, JobError
     public static Task UpdatePullRequest(this IApiHandler handler, UpdatePullRequest updatePullRequest) => handler.PostAsJson("update_pull_request", updatePullRequest);
     public static Task MarkAsProcessed(this IApiHandler handler, MarkAsProcessed markAsProcessed) => handler.PatchAsJson("mark_as_processed", markAsProcessed);
 
-    private static Task PostAsJson(this IApiHandler handler, string endpoint, object body) => handler.SendAsync(endpoint, body, "POST");
-    private static Task PatchAsJson(this IApiHandler handler, string endpoint, object body) => handler.SendAsync(endpoint, body, "PATCH");
+    private static Task PostAsJson(this IApiHandler handler, string endpoint, object body) => handler.WithRetries(() => handler.SendAsync(endpoint, body, "POST"));
+    private static Task PatchAsJson(this IApiHandler handler, string endpoint, object body) => handler.WithRetries(() => handler.SendAsync(endpoint, body, "PATCH"));
+
+    private const int MaxRetries = 3;
+    private const int MinRetryDelay = 3;
+    private const int MaxRetryDelay = 10;
+
+    private static async Task WithRetries(this IApiHandler handler, Func<Task> action)
+    {
+        var retryCount = 0;
+        while (true)
+        {
+            try
+            {
+                await action();
+                return;
+            }
+            catch (HttpRequestException) when (retryCount < MaxRetries)
```

Comment:
seems this will retry everything? Can we try retrying only on known retryable errors?

Reply:
The instance in the logs that I found, I have to keep this broad to catch that.  Specifically, there was an internal `HttpIOException` with the status `ResponseEnded` which resulted in a `null` status code, so I couldn't restrict this to just `5xx` errors or the like because the .NET API didn't return a status code.

Reply:
I've been thinking more about this.  What if we retry only if the `StatusCode` is `null` (meaning something happened to the connection), _OR_ the `StatusCode` is `5xx`, indicating a problem on the server?  That way we're not retrying any `400`s?